vault/expiration.go

```diff
@@ -943,6 +956,27 @@ func (m *ExpirationManager) lazyRevokeInternal(ctx context.Context, leaseID stri
 	return nil
 }
 
+// TODO should context passed to revokeCommon contain the lease namespace?
+// should be run on a schedule. something like once a day, maybe once a week
+func (m *ExpirationManager) attemptIrrevocableLeasesRevoke() {
+	m.irrevocable.Range(func(k, v interface{}) bool {
+		leaseID := k.(string)
+		le := v.(*leaseEntry)
+
+		if le.ExpireTime.Add(7 * 24 * time.Hour).Before(time.Now()) {
+			// if the lease has expired at least a week ago, try again
+			if err := m.revokeCommon(context.Background(), leaseID, false, false); err != nil {
```

Comment:
Can we derive a context from a.activeContext and add a timeout here?

Reply:
how much timeout would you recommend? 1 minute?

Reply:
6197b3c04

let me know what you think about the 1 minute timeout

Reply:
I think 1m is fine since this is the retry code.